builder/frameworks/espidf.py

```
@@ -2171,15 +2106,15 @@ def idf_lib_copy(source, target, env):
 if "espidf" in env.subst("$PIOFRAMEWORK") and (flag_custom_component_add == True or flag_custom_component_remove == True):
     def idf_custom_component(source, target, env):
         try:
-            shutil.copy(join(ARDUINO_FRAMEWORK_DIR,"idf_component.yml.orig"),join(ARDUINO_FRAMEWORK_DIR,"idf_component.yml"))
+            shutil.copy(str(Path(ARDUINO_FRAMEWORK_DIR) / "idf_component.yml.orig"), str(Path(ARDUINO_FRAMEWORK_DIR) / "idf_component.yml"))
             print("*** Original Arduino \"idf_component.yml\" restored ***")
         except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)